content/50_deploying_argocd_to_eks/20_create_a_namespace.md

```diff
@@ -0,0 +1,57 @@
++++
+title = "Creating a Namespace"
+chapter = false
+weight = 20
++++
+
+Next, declare a [namespace object](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/).
+This will allow scoping the deployment of the ArcoCD components to that namespace.
+
+To do this, we need to create a stack reference to the project where we created the eks infrastructure so we can get the
+Kubeconfig and be able to build the correct kubernetes provider. We are going to make the stack reference name configurable:
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+
+const pulumiConfig = new pulumi.Config();
+
+// Existing Pulumi stack reference in the format:
+// <organization>/<project>/<stack> e.g. "myUser/myProject/dev"
+const clusterStackRef = new pulumi.StackReference(pulumiConfig.require("clusterStackRef"));
+```
+
+Now we can get the kubeconfig from the eks cluster for use in our provider. Append this to your `index.ts` file:
+
+```typescript
+const provider = new k8s.Provider("k8s", { kubeconfig: eks.getOutput("kubeconfig") });
+```
+
+Now we can create the namespace using the provider we just created. Append this to your `index.ts` file:
+
+```typescript
+let name = "argocd"
```

Comment:
we generally use `const` for static values instead of `let`

Reply:
👌